guides-markdown/bcoin-clock.md

```diff
@@ -0,0 +1,560 @@
+
+```bash
+$ cat ~/blocks/526358
+{
+	"hash":"00000000000000000009392f74e26fce58359267a64aa9aa86708062a2296492",
+	"version":536870912,
+	"prevBlock":"0000000000000000001d437becc92a9cdf38af11105bf6370c6cde001bc0a2ec",
+	"merkleRoot":"bbbb711d2910205c8c2b444572e79ec5812705203103afc5f82980cb505550e1",
+	"time":1528332683,
+	"bits":389609537,
+	"nonce":3638200179,
+	"totalTX":1199,
+	"hashes":[
+		"bbbb711d2910205c8c2b444572e79ec5812705203103afc5f82980cb505550e1"
+	],
+	"flags":"00"
+}
+```
+
+## Enter: Python
+
+Congratulations! If you made it this far, you have a working Bitcoin light client in the palm of your hand, ready to send and receive money. You've modified it to record details about each new block in a file on disk. The rest of this guide will focus on interacting with the those block files, as well as the bcoin node and wallet servers via cURL requests in python. We use Python 2.7 because it is already installed on Raspbian, easy to interact with the command line, and because the extra hardware we add later is driven by libraries written in python.
+
+*Reminder: The finished script is available at [https://github.com/pinheadmz/bcoin-clock](https://github.com/pinheadmz/bcoin-clock)*
+
+First install some dependencies:
+
+```bash
+sudo apt-get install python-pip python-pil python-requests
+sudo pip install pyqrcode
+```
+
+Create a new python file `gui_clock.py` and start with the code below. Make sure bcoin is already running with the `spv_clock.js` script, and then run your new python script with the command `python gui_clock.py`. If all goes well you should see a long JSON output, followed by a Bitcoin address and QR code, right in the terminal!
+
+```python
+import requests
+import pyqrcode
+import json
+
+### default API endpoints, not using any authentication key
+# testnet
+#nodeurl = "http://127.0.0.1:18332/"
+#wallurl = "http://127.0.0.1:18334/wallet/primary/"
+# main
+nodeurl = "http://127.0.0.1:8332/"
+wallurl = "http://127.0.0.1:8334/wallet/primary/"
+
+### set up some functions to get data from bcoin node & wallet servers
+### these functions are equivalent to using cURL from the command line
+### the Python requests package makes this super easy for us
+def getInfo():
+	return requests.get(nodeurl).json()
+
+def getAddress():
+	params = {"account":"default"}
+	return requests.post(wallurl + 'address', json=params).json()
+
+def getBalance():
+	return requests.get(wallurl + 'balance?account=default').json()
+	
+### Output!
+print json.dumps(getInfo(), indent=1)
+print json.dumps(getBalance(), indent=1)
+
+# display address and QR code
+addr = getAddress()['address']
+code = pyqrcode.create(addr, 'M', version=3).terminal(quiet_zone=1)
```

Comment:
might be nice to have a description of what this is (like what the options are for), and/or maybe a link for more info.